### DISCUSSION TITLE
Search function not working in second docs page

### DISCUSSION THREAD

**Discussion by xbufu** [thuliteio/doks#543] [Q&A] [answered]
Hi!

I am trying to create a second docs page, e.g. called `docs`, that should have the same layout as the `docs` page. I managed to do this by following [this](https://github.com/h-enk/doks/discussions/173) discussion topic. I managed to add the second page, all with correct menu entries and the sidebar showing correctly. I have also managed to get the search bar to appear correctly now, by adding the appropriate entries in `/layouts/partials/header/header.html`.

My problem has now become that the search does not return any results. There are no suggestions showing up while I type and when I hit enter it just reloads the page.

So far I have not found the reason for this. Can anyone help me fix it? I have published the source code here: https://github.com/xbufu/wiki.git and published them here: https://wiki.bufu-sec.com/

**Comment by h-enk** [comment, marked as answer]
For a similar setup, see: https://github.com/h-enk/doks/discussions/395

Also, make sure to update `./layouts/partials/footer/script-footer.html` (lines 77 +102), make it something like (not tested):

```html
{{ if and (.Site.Params.options.flexSearch) (site.RegularPages "Type" "in" site.Params.mainSections) -}}
```

**Reply by xbufu** [reply]
Thanks a lot, I got it working now! Although I simply made the same edit to `./layouts/partials/footer/script-footer.html` as I did to `./layouts/partials/header/header.html`. I replaced lines 77 and 102 with:

`{{ if and (.Site.Params.options.flexSearch) (eq .Section "docs" "wiki") -}}`

Think this is good for now, although I was looking into using Algolia for the search as well, since getdoks.org uses it, but that seems too complicated for me for now :D

Update:

I created a test repo and added the second section, everything works! If anyone wants to reproduce it, you can find the commit [here](https://github.com/xbufu/test-wiki/commit/8d4ff9b089a618e9319de1f3e986f87acd609417).

**Comment by james-d-elliott** [comment]
For those interested a working modification (at least for me) using the above examples is as follows:

```diff
Index: layouts/partials/footer/script-footer.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/layouts/partials/footer/script-footer.html b/layouts/partials/footer/script-footer.html
--- a/layouts/partials/footer/script-footer.html	(revision f5d1bae87c4fba564bcfea69bf6cdbf14332f8ea)
+++ b/layouts/partials/footer/script-footer.html	(date 1638266546293)
@@ -74,7 +74,7 @@
   {{ with .Params.mermaid -}}
     <script src="{{ $mermaid.RelPermalink }}" defer></script>
   {{ end -}}
-  {{ if and (.Site.Params.options.flexSearch) (eq .Section "docs") -}}
+  {{ if and (.Site.Params.options.flexSearch) (in site.Params.mainSections .Section) -}}
     <script src="{{ $index.RelPermalink }}" defer></script>
   {{ end -}}
 {{ else -}}
@@ -99,7 +99,7 @@
   {{ with .Params.mermaid -}}
     <script src="{{ $mermaid.RelPermalink }}" integrity="{{ $mermaid.Data.Integrity }}" crossorigin="anonymous" defer></script>
   {{ end -}}
-  {{ if and (.Site.Params.options.flexSearch) (eq .Section "docs") -}}
+  {{ if and (.Site.Params.options.flexSearch) (in site.Params.mainSections .Section) -}}
     <script src="{{ $index.Permalink }}" integrity="{{ $index.Data.Integrity }}" crossorigin="anonymous" defer></script>
   {{ end -}}
-{{ end -}}
\ No newline at end of file
+{{ end -}}
Index: layouts/partials/header/header.html
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/layouts/partials/header/header.html b/layouts/partials/header/header.html
--- a/layouts/partials/header/header.html	(revision f5d1bae87c4fba564bcfea69bf6cdbf14332f8ea)
+++ b/layouts/partials/header/header.html	(date 1638266785900)
@@ -93,7 +93,7 @@
 </div>
 {{ end -}}
 
-{{ if eq .Section "docs" -}}
+{{ if in site.Params.mainSections .Section -}}
 <nav class="doks-subnavbar py-2 sticky-lg-top{{ if eq .Site.Params.options.flexSearch false }} d-lg-none{{ end }}" aria-label="Secondary navigation">
   <div class="container-xxl d-flex align-items-md-center">
 
Index: assets/js/index.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/assets/js/index.js b/assets/js/index.js
--- a/assets/js/index.js	(revision f5d1bae87c4fba564bcfea69bf6cdbf14332f8ea)
+++ b/assets/js/index.js	(date 1638266164341)
@@ -94,7 +94,7 @@
 
   // https://discourse.gohugo.io/t/range-length-or-last-element/3803/2
 
-  {{ $list := (where .Site.Pages "Section" "docs") -}}
+  {{ $list := (where .Site.Pages "Section" "configuration") -}}
   {{ $len := (len $list) -}}
 
   index.add(
```
